bpython/test/test_line_properties.py

```diff
@@ -178,10 +178,19 @@ def test_simple(self):
         self.assertAccess("asdf[<(>|]")
         self.assertAccess("asdf[<(1>|]")
         self.assertAccess("asdf[<(1,>|]")
+        self.assertAccess("asdf[<(1,)>|]")
         self.assertAccess("asdf[<(1, >|]")
         self.assertAccess("asdf[<(1, 2)>|]")
         # TODO self.assertAccess('d[d[<12|>')
         self.assertAccess("d[<'a>|")
+        self.assertAccess("object.dict['a'bcd'], object.dict[<'abc>|")
+        self.assertAccess(r"object.dict[<'a\'\\\"\n\\'>|")
+        self.assertAccess("object.dict[<\"abc'>|")
+        self.assertAccess("object.dict[<(1, 'apple', 2.134>|]")
+        self.assertAccess("object.dict[<(1, 'apple', 2.134)>|]")
+        self.assertAccess("object.dict[<-1000>|")
+        self.assertAccess("object.dict[<-0.23948>|")
+        self.assertAccess("object.dict[<'\U0001ffff>|")
```

Comment:
I think it would make sense to also add some tests for string keys containing `[` and `]` .

Reply:
I misinterpreted this and added a test where the user is typing a string dict key inside `[` and `]` and found an issue so I fixed that by modifying the regex. Also added some tests for strings that contain `[` and `]`.